include/swift/Frontend/ModuleInterfaceSupport.h

```diff
@@ -42,23 +41,26 @@ struct ModuleInterfaceOptions {
   /// [TODO: Clang-type-plumbing] This check should go away.
   bool PrintFullConvention = false;
 
-  /// Copy of all the command-line flags passed at .swiftinterface
-  /// generation time, re-applied to CompilerInvocation when reading
-  /// back .swiftinterface and reconstructing .swiftmodule.
-  std::string Flags;
+  struct InterfaceFlags {
+    /// Copy of all the command-line flags passed at .swiftinterface
+    /// generation time, re-applied to CompilerInvocation when reading
+    /// back .swiftinterface and reconstructing .swiftmodule.
+    std::string Flags = "";
 
-  /// Keep track of flags to be printed in package.swiftinterface only.
-  /// If -disable-print-package-name-for-non-package-interface is passed,
-  /// package-name flag should only be printed in package.swiftinterface.
-  std::string FlagsForPackageOnly;
+    /// Flags that should be emitted to the .swiftinterface file but are OK to
+    /// be ignored by the earlier version of the compiler.
+    std::string IgnorableFlags = "";
+  };
 
-  /// Flags that should be emitted to the .swiftinterface file but are OK to be
-  /// ignored by the earlier version of the compiler.
-  std::string IgnorableFlags;
+  /// Flags which appear only in all .swiftinterface files.
+  InterfaceFlags PublicFlags = {};
 
-  /// Ignorable flags that should only be printed in .private.swiftinterface file;
-  /// e.g. -package-name PACKAGE_ID
-  std::string IgnorablePrivateFlags;
+  /// Flags which appear in the private and package .swiftinterface files, but
+  /// not the public interface.
+  InterfaceFlags PrivateFlags = {};
```

Comment:
Shouldn't this be PrivateAndPackageFlags or NonPublicFlags?

Reply:
By the same logic, `PublicFlags` would need to be named something like `PublicAndPrivateAndPackageFlags`. I'd rather not use names like that here. To me, it's a given that the private interface is a superset of the public interface, and the package interface is a superset of the private interface.

Reply:
Ok that's a fair point.